lib/datadog/core/configuration/option.rb

```diff
@@ -9,17 +9,36 @@ class Option
         attr_reader \
           :definition
 
+        # Option setting precedence. Higher number means higher precedence.
+        module Precedence
+          REMOTE_CONFIGURATION = 2
+          PROGRAMMATIC = 1
+          DEFAULT = 0
```

Comment:
Perhaps worth documenting that "default" here includes defaults in the code as well as environment variables (e.g. there's two levels that are somewhat collapsed into one, that we don't explicitly represent).

Reply:
Done!